src/v/serde/rw/iobuf.h

```diff
@@ -7,6 +7,7 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0
 
+// IWYU pragma: always_keep; symbols for ADL
```

Comment:
ahh, yeh i have an idea about how to fix this. something like build a wrapper type that is defined in these headers then force `serde_fields` to use the wrapper. then we'd trick clangd into knowing which header was needed. maybe there are other ways